src/components/BSelect/BSelect.vue

```diff
@@ -0,0 +1,284 @@
+          <span class="dropdown-arrow"><i class="fa fa-caret-down" aria-hidden="true"></i></span>
+        </li>
+        <span v-show="selectErrors.has('selectedOption')" class="error" data-vv-as="selectLabel">{{ selectErrors.first('selectedOption') }}</span>
+        <li
+          v-if=opened
+          v-for="option in options"
+          :aria-selected="selectedOption && selectedOption.id === option.id"
+          :id="'option' + option.id"
+          :key="'selectOption' + option.id"
+          :value=option.id
+          @click="selectOption(option)"
+          @keyup.38.prevent="upHandler"
+          @keyup.40.prevent="downHandler"
+          @keyup.13.prevent="selectOption(option)"
+          @keyup.32.prevent.stop="selectOption(option)"
+          class="li-opened options"
+          role="option"
+          tabindex='-1'
+        > {{ option.text }}
+        </li>
+      </ul>
+    </div>
+  </div>
+</template>
+
+<script>
+
+import { Validator } from 'vee-validate'
+
+const options = []
+
+export default {
+  name: 'b-select',
+  validator: null,
+  props: {
+
+    id: {
+      type: String,
+      required: true
+    },
+    selectOptions: {
+      type: Array
+    },
+    isRequired: {
+      type: Boolean,
+      required: true
+    },
+    selectLabel: {
+      type: String,
+      required: true
+    },
+    initialValue: {
+      type: Object,
+      required: false
+    },
+    placeholder: {
+      type: String,
+      required: false,
+      default: 'Choose Option'
+    }
+  },
+
+  methods: {
+
+    /**
+     * Shows and hides line options for the select list.
+     */
+
+    toggleList () {
+      this.opened = !this.opened
+      if (this.opened) {
+        this.$nextTick( function () {
+          this.setFocus()
+        })
+      } else {
+        if (this.isRequired) {
+          this.validate()
+        }
+        document.querySelector(`#dropdown${this.id}`).focus()
+      }
+    },
+
+    /**
+     * @param {Object} option
+     * Handles the selection of a line and closed the list.
+     */
+
+    selectOption (option) {
+      this.selectedOption = option
+      this.opened = !this.opened
+      if (this.isRequired) {
+        this.validate()
+      }
+      document.querySelector(`#dropdown${this.id}`).focus()
+    },
+
+    /**
+     * Handles validation if required.
+     */
+
+    validate () {
+      this.validator.validate('selectedOption', this.selectedOption)
+    },
+
+    /**
+     * Handles the focus setting for accessibility purposes
+     */
+
+    setFocus () {
+      if (this.selectedOption) {
+        const line = document.querySelector(`#${this.id} #option${this.selectedOption.id}`)
+        line.focus()
+      } else {
+        const line = document.querySelector(`#${this.id} #option${this.options[0].id}`)
+        line.focus()  
+      }
+    },
+
+    /**
+     * Handles the up arrow (38) key press event
+     */
+
+    upHandler () {
+      const target = event.target;
```

Comment:
Where is event coming from if it's not passed into the method (which I think it should be)?

Reply:
not sure what you are asking here

Reply:
`event` doesn't appear to be defined anywhere because it's not a parameter for the method.

Reply:
oh i see what you are asking.  Vue magic, but I'll update it to be explicit.